### ISSUE TITLE
rclone hides failure on AWS S3 bucket creation

### ISSUE THREAD

**Issue by saffroy** [rclone/rclone#7351]
#### The associated forum post URL from `https://forum.rclone.org`

None.

#### What is the problem you are having with rclone?

When trying to create an Amazon S3 bucket, I get no error when I should get one.

#### What is your rclone version (output from `rclone version`)

```
$ ./rclone selfupdate --beta
2023/10/05 17:23:29 NOTICE: Successfully updated rclone from version v1.64.0 to version v1.65.0-beta.7409.a75256384

$ ./rclone version 
rclone v1.65.0-beta.7409.a75256384
- os/version: debian 12.1 (64 bit)
- os/kernel: 6.1.0-12-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.1
- go/linking: static
- go/tags: none
```

#### Which cloud storage system are you using? (e.g. Google Drive)

AWS S3

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

I have no bucket named `testtest`, and tried to create one:
```
$ ./rclone lsd s3: |grep test

$ ./rclone mkdir s3:testtest ; echo $?
0
```

But nothing happened:

```
$ ./rclone lsd s3: |grep test 
```

It fails silently, actually, despite receiving a 409 HTTP status code:
```
$ ./rclone -vvv --dump headers mkdir s3:testtest
2023/10/05 17:30:53 DEBUG : rclone: Version "v1.65.0-beta.7409.a75256384" starting with parameters ["./rclone" "-vvv" "--dump" "headers" "mkdir" "s3:testtest"]
2023/10/05 17:30:53 DEBUG : Creating backend with remote "s3:testtest"
2023/10/05 17:30:53 DEBUG : Using config file from "/home/saffroy/.config/rclone/rclone.conf"
2023/10/05 17:30:53 DEBUG : You have specified to dump information. Please be noted that the Accept-Encoding as shown may not be correct in the request and the response may not show Content-Encoding if the go standard libraries auto gzip encoding was in effect. In this case the body of the request will be gunzipped before showing it.
2023/10/05 17:30:53 DEBUG : You have specified to dump information. Please be noted that the Accept-Encoding as shown may not be correct in the request and the response may not show Content-Encoding if the go standard libraries auto gzip encoding was in effect. In this case the body of the request will be gunzipped before showing it.
2023/10/05 17:30:53 DEBUG : S3 bucket testtest: Making directory
2023/10/05 17:30:53 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2023/10/05 17:30:53 DEBUG : HTTP REQUEST (req 0xc0005eb900)
2023/10/05 17:30:53 DEBUG : PUT / HTTP/1.1
Host: testtest.s3.eu-west-3.amazonaws.com
User-Agent: rclone/v1.65.0-beta.7409.a75256384
Content-Length: 153
Authorization: XXXX
X-Amz-Content-Sha256: a0fa36eb3d33c2b2ff7f36bf054cae8281ee0b23b4309a590595a1817f636196
X-Amz-Date: 20231005T153053Z
Accept-Encoding: gzip

2023/10/05 17:30:53 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2023/10/05 17:30:54 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2023/10/05 17:30:54 DEBUG : HTTP RESPONSE (req 0xc0005eb900)
2023/10/05 17:30:54 DEBUG : HTTP/1.1 409 Conflict
Transfer-Encoding: chunked
Content-Type: application/xml
Date: Thu, 05 Oct 2023 15:30:53 GMT
Server: AmazonS3
X-Amz-Bucket-Region: us-east-1
X-Amz-Id-2: [REDACTED]
X-Amz-Request-Id: JAA1SES5NBR4VTPT

2023/10/05 17:30:54 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2023/10/05 17:30:54 DEBUG : 6 go routines active
```

The reason for the error is this oddity:

> The following rules apply for naming buckets in Amazon S3:
> ...
> A bucket name cannot be used by another AWS account in the same partition until the bucket is deleted.

Source https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

Creation of a bucket with a unique name does work, so it is not a configuration issue:
```
$ ./rclone lsd s3: |grep test
$ ./rclone mkdir s3:test-$RANDOM
$ ./rclone lsd s3: |grep test
          -1 2023-10-05 17:51:52        -1 test-19842
```

A different tool gave an error, and a very helpful one:

```
$ s3cmd mb s3://testtest
ERROR: Bucket 'testtest' already exists
ERROR: S3 error: 409 (BucketAlreadyExists): The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.

$ echo $?
13
```

**Comment by ncw** [comment]
> ERROR: S3 error: 409 (BucketAlreadyExists): The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.

Unhelpfully some non s3 systems give the same error code `BucketAlreadyExists` if you own the bucket or not.

Rclone tries to make the bucket before doing operations on it, so if the bucket already exists it ignore the `BucketAlreadyExists` error.

The reason rclone does this is probably a design mistake that I haven't removed as I don't want to break backwards compatibility.

However s3 does now (it didn't used to) give back the error BucketAlreadyOwnedByYou and we know we are using s3 because of the provider field, so that suggests a fix - only check  BucketAlreadyOwnedByYou on S3.

I've added a provider quirk and and coded that up - I think this should give an error for you now.

[v1.65.0-beta.7410.3a5a870c9.fix-7351-s3-bucket-exists](https://beta.rclone.org/branch/fix-7351-s3-bucket-exists/v1.65.0-beta.7410.3a5a870c9.fix-7351-s3-bucket-exists/) on branch [fix-7351-s3-bucket-exists](https://github.com/rclone/rclone/tree/fix-7351-s3-bucket-exists) (uploaded in 15-30 mins)

**Comment by saffroy** [comment]
A quick test:
```
$ ./rclone version
rclone v1.65.0-beta.7410.3a5a870c9.fix-7351-s3-bucket-exists
- os/version: debian 12.1 (64 bit)
- os/kernel: 6.1.0-12-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.1
- go/linking: static
- go/tags: none

$ ./rclone lsd s3:|grep test

$ ./rclone mkdir s3:testtest
2023/10/06 18:05:27 ERROR : Attempt 1/3 failed with 1 errors and: BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
        status code: 409, request id: MHVRRCDN6GM5GNA2, host id: Jd+Py7ByJ+v81eWivWNWXC5xLutfb19Lc5XfAwr0u6oR8h1WEOiAh1cfqpoH9rcnS2EZZn9cF9s=
2023/10/06 18:05:27 ERROR : Attempt 2/3 failed with 1 errors and: BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
        status code: 409, request id: MHVQFB8Q025PW2JQ, host id: 9F0p2dnpBhE5uPAC1LvhRAy90jxbW91QTtg7moNMsur4HYfxzyJu/lHUexRYtzYS3K+GeLXo1mw=
2023/10/06 18:05:27 ERROR : Attempt 3/3 failed with 1 errors and: BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
        status code: 409, request id: 6XRCXSBW6FG6QA85, host id: +0SKfuwfyR55gWSZwrbz5WhEoHX8QAR2GIFcgx5AWA7ZEeQ7lyVqjzbV+FHdUHMaFJ7bCszI6L4=
2023/10/06 18:05:27 Failed to mkdir: BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
        status code: 409, request id: 6XRCXSBW6FG6QA85, host id: +0SKfuwfyR55gWSZwrbz5WhEoHX8QAR2GIFcgx5AWA7ZEeQ7lyVqjzbV+FHdUHMaFJ7bCszI6L4=

$ echo $?
1
```

So, it works, good.

But I don't see why there are retries in this case, the status is clear enough that reries won't help: is that expected?

**Comment by saffroy** [comment]
It doesn't retry needlessly anymore:

```
$ ./rclone version
rclone v1.65.0-beta.7417.8ca56ae1b.fix-7351-s3-bucket-exists
- os/version: debian 12.2 (64 bit)
- os/kernel: 6.1.0-12-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.1
- go/linking: static
- go/tags: none

$ ./rclone mkdir s3:testtest
2023/10/09 18:50:18 ERROR : Can't retry any of the errors - not attempting retries
2023/10/09 18:50:18 Failed to mkdir: BucketAlreadyExists: The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.
        status code: 409, request id: HAHYJJ4JD6XN7D9W, host id: UYjIaEBmgfMZdCIbgukKKxeMO26S2xkKU0+lnSEHSNyW2SLJ66HFRL4pFyf0uVLYcD2ymWU2TkbGdxtn9Sm1YQ==
```

It *is* a bit noisy though. Why mention "can't retry", given that retries wouldn't help here?

Just a guess, but maybe there is a more general need in rclone commands to qualify when retries are needed:

```
$ ./rclone mkdir /foobar
2023/10/09 19:00:54 ERROR : Attempt 1/3 failed with 1 errors and: mkdir /foobar: permission denied
2023/10/09 19:00:54 ERROR : Attempt 2/3 failed with 1 errors and: mkdir /foobar: permission denied
2023/10/09 19:00:54 ERROR : Attempt 3/3 failed with 1 errors and: mkdir /foobar: permission denied
2023/10/09 19:00:54 Failed to mkdir: mkdir /foobar: permission denied

$ ./rclone touch /foobar
2023/10/09 19:00:58 ERROR : Attempt 1/3 failed with 1 errors and: failed to touch (create): open /foobar: permission denied
2023/10/09 19:00:58 ERROR : Attempt 2/3 failed with 1 errors and: failed to touch (create): open /foobar: permission denied
2023/10/09 19:00:58 ERROR : Attempt 3/3 failed with 1 errors and: failed to touch (create): open /foobar: permission denied
2023/10/09 19:00:58 Failed to touch: failed to touch (create): open /foobar: permission denied

```

That would be a discussion for another issue though.

I have no other feedback, so feel free to close this one. Thanks!

**Comment by ncw** [comment]
> It doesn't retry needlessly anymore:

Good :-)

> It _is_ a bit noisy though. Why mention "can't retry", given that retries wouldn't help here?

Rclone normally retries everything, so not retrying is unusual!

> I have no other feedback, so feel free to close this one. Thanks!

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.65
